pkg/virt-launcher/virtwrap/access-credentials/access_credentials_test.go

```diff
@@ -126,6 +126,105 @@ var _ = Describe("AccessCredentials", func() {
+		mockDomain.EXPECT().Free().Times(1)
+
+		expectedOpenCmd := fmt.Sprintf(`{"execute": "guest-file-open", "arguments": { "path": "%s/authorized_keys", "mode":"r" } }`, filePath)
+		expectedWriteOpenCmd := fmt.Sprintf(`{"execute": "guest-file-open", "arguments": { "path": "%s/authorized_keys", "mode":"w" } }`, filePath)
+		expectedOpenCmdRes := `{"return":1000}`
+
+		existingKey := base64.StdEncoding.EncodeToString([]byte("ssh some existing key"))
+		expectedReadCmd := `{"execute": "guest-file-read", "arguments": { "handle": 1000 } }`
+		expectedReadCmdRes := fmt.Sprintf(`{"return":{"count":24,"buf-b64": "%s"}}`, existingKey)
+
+		mergedKeys := base64.StdEncoding.EncodeToString([]byte(strings.Join(authorizedKeys, "\n")))
+		expectedWriteCmd := fmt.Sprintf(`{"execute": "guest-file-write", "arguments": { "handle": 1000, "buf-b64": "%s" } }`, mergedKeys)
+
+		expectedCloseCmd := `{"execute": "guest-file-close", "arguments": { "handle": 1000 } }`
+
+		expectedExecReturn := `{"return":{"pid":789}}`
+		expectedStatusCmd := `{"execute": "guest-exec-status", "arguments": { "pid": 789 } }`
+
+		getentBase64Str := base64.StdEncoding.EncodeToString([]byte("someowner:x:1111:2222:Some Owner:/home/someowner:/bin/bash"))
+		expectedHomeDirCmd := `{"execute": "guest-exec", "arguments": { "path": "getent", "arg": [ "passwd", "someowner" ], "capture-output":true } }`
+		expectedHomeDirCmdRes := fmt.Sprintf(`{"return":{"exitcode":0,"out-data":"%s","exited":true}}`, getentBase64Str)
+
+		expectedMkdirCmd := fmt.Sprintf(`{"execute": "guest-exec", "arguments": { "path": "mkdir", "arg": [ "-p", "%s" ], "capture-output":true } }`, filePath)
+		expectedMkdirRes := `{"return":{"exitcode":0,"out-data":"","exited":true}}`
+
+		expectedParentChownCmd := fmt.Sprintf(`{"execute": "guest-exec", "arguments": { "path": "chown", "arg": [ "1111:2222", "%s" ], "capture-output":true } }`, filePath)
+		expectedParentChownRes := `{"return":{"exitcode":0,"out-data":"","exited":true}}`
+
+		expectedParentChmodCmd := fmt.Sprintf(`{"execute": "guest-exec", "arguments": { "path": "chmod", "arg": [ "700", "%s" ], "capture-output":true } }`, filePath)
+		expectedParentChmodRes := `{"return":{"exitcode":0,"out-data":"","exited":true}}`
+
+		expectedFileChownCmd := fmt.Sprintf(`{"execute": "guest-exec", "arguments": { "path": "chown", "arg": [ "1111:2222", "%s/authorized_keys" ], "capture-output":true } }`, filePath)
+		expectedFileChownRes := `{"return":{"exitcode":0,"out-data":"","exited":true}}`
+
+		expectedFileChmodCmd := fmt.Sprintf(`{"execute": "guest-exec", "arguments": { "path": "chmod", "arg": [ "600", "%s/authorized_keys" ], "capture-output":true } }`, filePath)
+		expectedFileChmodRes := `{"return":{"exitcode":0,"out-data":"","exited":true}}`
+
+		// Detect user home dir
+		mockConn.EXPECT().QemuAgentCommand(expectedHomeDirCmd, domName).Return(expectedExecReturn, nil)
```

Comment:
Makes sense to add `Times(1)` here (and following lines) to capture the exact behavior?

Reply:
done